crates/typst/src/eval/value.rs

```diff
@@ -69,6 +71,43 @@ pub enum Value {
     Dyn(Dynamic),
 }
 
+impl Serialize for Value {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Value::None => serializer.serialize_none(),
+            Value::Auto => serializer.serialize_str("auto"),
+            Value::Bool(v) => serializer.serialize_bool(*v),
+            Value::Int(v) => serializer.serialize_i64(*v),
+            Value::Float(v) => serializer.serialize_f64(*v),
+            Value::Str(v) => serializer.serialize_str(v),
+            Value::Bytes(v) => serializer.serialize_bytes(v),
+            Value::Array(v) => {
+                let mut seq = serializer.serialize_seq(Some(v.len()))?;
```

Comment:
In general, the serializations should probably be separate impls on the respective types.

Reply:
Yes, but maybe the serialization would produce unnecessary structure to map the types nested in the enum variants. So for the first, I chose to directly serialize Value.

Reply:
I don't understand. Isn't delegating to a sub-serialize part of how serde works?

Reply:
See https://serde.rs/enum-representations.html ... I wasn't aware of the untagged variant and wanted to avoid explicit enum variant tagging.